pkg/util/wsstream/conn.go

```diff
@@ -154,6 +165,29 @@ func NewConn(channels ...ChannelType) *Conn {
 	return conn
 }
 
+func generateProtocols(prefixes []string, suffixes []string) []string {
+	if len(prefixes) == 0 {
+		prefixes = []string{""}
+	}
+	protos := make([]string, 0, len(suffixes)*len(prefixes))
+	for _, prefix := range prefixes {
+		if prefix != "" {
+			if errs := validation.IsDNS1123Subdomain(prefix); len(errs) > 0 {
+				glog.Fatalf("websocket protocol prefix %q is no DNS 1123 subdomain: %v", prefix, errs)
```

Comment:
Don't use glog.Fatal anywhere outside of CLI code or a very limited set of init code, and even then sparingly